src/main/java/org/magicdgs/thaplv/utils/AlleleUtils.java

```diff
@@ -0,0 +1,87 @@
+ * and the following disclaimer in the documentation and/or other materials provided with the
+ * distribution.
+ *
+ * 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse
+ * or promote products derived from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR
+ * IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR
+ * CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+ * THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+package org.magicdgs.thaplv.utils;
+
+import htsjdk.variant.variantcontext.Allele;
+import htsjdk.variant.variantcontext.VariantContext;
+
+import java.util.HashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Stream;
+
+/**
+ * Static utils for {@link htsjdk.variant.variantcontext.Genotype} and {@link VariantContext} for
+ * haploid individuals.
+ *
+ * WARNING: only works for haploid/homozygous individuals, not taking into account the second
+ * allele when parsing genotypes
+ *
+ * @author Daniel Gomez-Sanchez (magicDGS)
+ */
+public class AlleleUtils {
+
+    private AlleleUtils() { }
+
+    /**
+     * Get the allele counts from an allele stream
+     *
+     * @param alleles       alleles to count
+     * @param includeNoCall include in the counts the no call allele
+     *
+     * @return map of alleles to counts
+     */
+    public static final HashMap<Allele, Integer> getAlleleCounts(final Stream<Allele> alleles,
+            final boolean includeNoCall) {
+        final HashMap<Allele, Integer> alleleCounts = new HashMap<>();
+        alleles.filter(a -> includeNoCall || a.isCalled()) // filter the no called
+                .forEach(al -> alleleCounts.merge(al, 1, Integer::sum));
+        return alleleCounts;
+    }
+
+    /**
+     * Get the allele frequencies from an allele stream
+     *
+     * @param alleles       alleles to get the allele frequencies from
+     * @param includeNoCall include in the counts the no call allele
+     *
+     * @return map of alleles to counts
+     */
+    public static final HashMap<Allele, Double> getAlleleFrequencies(final Stream<Allele> alleles,
+            final boolean includeNoCall) {
+        final HashMap<Allele, Double> alleleFreqs = new HashMap<>();
+        final AtomicInteger total = new AtomicInteger(0);
```

Comment:
Remove this `AtomicInteger` adding a `count()` in the lambda expression after merging in the map.